### PR TITLE
[Alignment] Deprecate `seqan3::alignment_coordinate`

### PR DESCRIPTION
**Pull request by marehr**
resolves https://github.com/seqan/product_backlog/issues/282

### CODE REVIEW

File: include/seqan3/alignment/matrix/all.hpp [hunk 1]
```diff
@@ -18,7 +18,7 @@
  * \see alignment
  */
 
-#include <seqan3/alignment/matrix/alignment_coordinate.hpp>
+#include <seqan3/alignment/matrix/detail/advanceable_alignment_coordinate.hpp>
 #include <seqan3/alignment/matrix/alignment_optimum.hpp>
 #include <seqan3/alignment/matrix/debug_matrix.hpp>
```
Comment (Irallia) [inline include/seqan3/alignment/matrix/all.hpp:23]:
```suggestion
#include <seqan3/alignment/matrix/alignment_optimum.hpp>
#include <seqan3/alignment/matrix/debug_matrix.hpp>
#include <seqan3/alignment/matrix/detail/advanceable_alignment_coordinate.hpp>
```

**Review by Irallia — CHANGES_REQUESTED** [review]
LGFM!
The only thing missing is the lexicographical order in the `all.hpp`

File: include/seqan3/alignment/matrix/all.hpp [hunk 2]
```diff
@@ -18,7 +18,7 @@
  * \see alignment
  */
 
-#include <seqan3/alignment/matrix/alignment_coordinate.hpp>
+#include <seqan3/alignment/matrix/detail/advanceable_alignment_coordinate.hpp>
```
Comment (marehr) [inline include/seqan3/alignment/matrix/all.hpp:21]:
```suggestion
```

File: include/seqan3/alignment/matrix/all.hpp (hunk 1 above)
Comment (marehr) [reply include/seqan3/alignment/matrix/all.hpp]: I removed it :)

**Review by Irallia — APPROVED** [review]

File: include/seqan3/alignment/matrix/all.hpp (hunk 2 above)
Comment (eseiler) [inline include/seqan3/alignment/matrix/all.hpp]: the all-header shouldn't include detail stuff?

File: test/unit/alignment/matrix/alignment_coordinate_test.cpp [hunk 3]
```diff
@@ -9,284 +9,10 @@
 
 #include <type_traits>
 
-#include <seqan3/alignment/matrix/detail/advanceable_alignment_coordinate.hpp>
+#include <seqan3/alignment/matrix/alignment_coordinate.hpp>
```
Comment (eseiler) [inline test/unit/alignment/matrix/alignment_coordinate_test.cpp]: Can you make this deprecation-warning safe?

File: include/seqan3/alignment/matrix/alignment_coordinate.hpp [hunk 4]
```diff
@@ -7,320 +7,20 @@
 
 /*!\file
  * \author Marcel Ehrhardt <marcel.ehrhardt AT fu-berlin.de>
- * \brief Provides seqan3::detail::alignment_coordinate.
+ * \brief Provides seqan3::alignment_coordinate.
  */
```
Comment (eseiler) [inline include/seqan3/alignment/matrix/alignment_coordinate.hpp:11]: The file itself should be deprecated and there should be a
```
SEQAN3_DEPRECATED_HEADER(
  "This header is deprecated and will be removed along all its content in SeqAn-3.1.0.")
```

Or will something survive in this header?

**Review by eseiler — CHANGES_REQUESTED** [review]
Some things regarding deprecation.
Also, a changelog entry is missing. (or is there already one?)

File: include/seqan3/alignment/matrix/alignment_coordinate.hpp (hunk 4 above)
Comment (marehr) [reply include/seqan3/alignment/matrix/alignment_coordinate.hpp:11]: No it won't. Do you know how the tests will handle it if I include a deprecated header?

Normally we would just include the new header, but in this case there is no new header. So I opted to not deprecate the header.

I could move the deprecated alignment_coordinate again into the other header and include that...

File: include/seqan3/alignment/matrix/all.hpp (hunk 2 above)
Comment (marehr) [reply include/seqan3/alignment/matrix/all.hpp]: Hmm I already removed that :)

**Review by eseiler — APPROVED** [review]

**Pull request merged**